public_html/wp-content/plugins/wordcamp-payments/includes/sponsor-payment-stripe.php

```diff
@@ -199,15 +199,23 @@ function _handle_post_data( &$data ) {
 				return;
 			}
 
+			try {
+				$decimal_amount = Stripe_Client::get_fractional_unit_amount( $currency, $amount );
+			} catch ( Exception $e ) {
+				$data['errors'][] = $e->getMessage();
+				return;
+			}
+
 			// Next step is to collect the card details via Stripe.
 			$data['step']    = STEP_PAYMENT_DETAILS;
 			$data['payment'] = array(
-				'payment_type' => $payment_type,
-				'wordcamp_id'  => $wordcamp_id,
-				'invoice_id'   => $invoice_id,
-				'description'  => $description,
-				'currency'     => $currency,
-				'amount'       => $amount,
+				'payment_type'   => $payment_type,
+				'wordcamp_id'    => $wordcamp_id,
+				'invoice_id'     => $invoice_id,
+				'description'    => $description,
+				'currency'       => $currency,
+				'amount'         => $amount,
+				'decimal_amount' => $decimal_amount,
```

Comment:
Why add the converted amount in a separate field instead of the normal `amount` field? Even if the Stripe API handles this correctly, I'm a bit leery of including both the converted and unconverted amounts in the payload.

Reply:
This isn't the payload sent to Stripe, this is just the data sent to the template – `amount` is still used in the display of the table:

![Screen Shot 2019-09-04 at 4 15 15 PM](https://user-images.githubusercontent.com/541093/64288089-36202680-cf2f-11e9-940a-073654a3f4c3.png)

Reply:
Oh, I see. Makes sense, carry on.